portal-web/test/functional/com/liferay/portalweb/tests/enduser/documentmanagement/documentsadministration/DMPermissions.testcase

```diff
@@ -369,6 +369,88 @@ definition {
 		MenuItem.viewNotPresent(menuItem = "Delete");
 	}
 
+	@description = "This is a test for LPS-155227. It checks that a user can not receive notifications without the subscribe permission of the folder."
+	@priority = 2
+	test CannotReceiveNotificationsWithoutFolderSubscriptionPermissions {
+		JSONGroup.addGroup(groupName = "Site Name");
+
+		JSONUser.addUser(
+			userEmailAddress = "userea@liferay.com",
+			userFirstName = "userfn",
+			userLastName = "userln",
+			userScreenName = "usersn");
+
+		JSONUser.setFirstPassword(
+			agreeToTermsAndAnswerReminderQuery = "true",
+			requireReset = "false",
+			userEmailAddress = "userea@liferay.com");
+
+		JSONUser.addUserToSite(
+			groupName = "Site Name",
+			userEmailAddress = "userea@liferay.com");
+
+		JSONLayout.addPublicLayout(
+			groupName = "Site Name",
+			layoutName = "Documents and Media Page");
+
+		JSONLayout.addWidgetToPublicLayout(
+			column = 2,
+			groupName = "Site Name",
+			layoutName = "Documents and Media Page",
+			widgetName = "Documents and Media");
+
+		JSONFolder.addFolder(
+			dmFolderDescription = "Parent Folder Description",
+			dmFolderName = "Parent Folder Name",
+			groupName = "Site Name");
+
+		JSONFolder.addFolder(
+			dmFolderDescription = "SubFolder Description",
+			dmFolderName = "SubFolder Name",
+			groupName = "Site Name",
+			parentFolderName = "Parent Folder Name");
+
+		DMNavigator.openToFolderInAdmin(
+			dmFolderName = "Parent Folder Name",
+			groupName = "Site Name",
+			siteURLKey = "site-name");
+
+		LexiconEntry.gotoEntryMenuItem(
+			menuItem = "Permissions",
+			rowEntry = "SubFolder Name");
+
+		Role.removePermissionPG(
+			permissionDefinitionKey = "INLINE_PERMISSIONS_SUBSCRIBE_CHECKBOX",
+			roleTitle = "Site Member");
+
+		JSONDocument.subscribeToFolder(
+			dmFolderName = "Parent Folder Name",
+			groupName = "Site Name",
+			userEmailAddress = "userea@liferay.com",
+			userPassword = "test");
+
+		Navigator.gotoSitePage(
+			pageName = "Documents and Media Page",
+			siteName = "Site Name");
+
+		DMNavigator.gotoFolder(dmFolderName = "Parent Folder Name");
+
+		DMNavigator.gotoFolder(dmFolderName = "SubFolder Name");
+
+		DMDocument.addPG(
+			dmDocumentDescription = "DM Image Description",
+			dmDocumentFile = "Document_1.jpg",
+			dmDocumentTitle = "DM Image Title");
```

Comment:
Please use JSONDocument.addFile for this

Reply:
I have tried to use JSON for the steps, but it can not trigger the notification.

Reply:
Can you at least combine the first three steps with DMNavigator.openToFolderInAdmin? I think it only requires the folder name to work

Reply:
Sure, but I need to update this macro DMNavigator.openToFolderInAdmin to make sure it can open a subfolder. OK, I will update it.